source/common/ssl/ssl_socket.h

```diff
@@ -69,25 +69,38 @@ class SslSocket : public Network::TransportSocket,
 
 class ClientSslSocketFactory : public Network::TransportSocketFactory {
 public:
-  ClientSslSocketFactory(const ClientContextConfig& config, Ssl::ContextManager& manager,
+  ClientSslSocketFactory(ClientContextConfigPtr config, Ssl::ContextManager& manager,
                          Stats::Scope& stats_scope);
 
   Network::TransportSocketPtr createTransportSocket() const override;
   bool implementsSecureTransport() const override;
 
 private:
+  // Will be called when config_ is changed
+  void onConfigUpdate();
```

Comment:
How this will be called? and tests? If the caller is not in this PR, include this in that PR.

Reply:
Done.  onConfigUpdate is removed.